From fed06578a116d903436d75d2662a7ce5c2f56d9d Mon Sep 17 00:00:00 2001
From: Aaron Brice <aaron@embeddedts.com>
Date: Wed, 2 Sep 2026 14:17:25 -0700
Subject: [PATCH] btattach: Update for glibc 2.42 compatability

glibc 2.42 baud rate constants changed, use cfsetspeed to be compatible
with newer glibc.

Upstream-Status: Pending
---
 tools/btattach.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/tools/btattach.c b/tools/btattach.c
index 5f7d19093..cdd7b125b 100644
--- a/tools/btattach.c
+++ b/tools/btattach.c
@@ -64,7 +64,12 @@ static int open_serial(const char *path, unsigned int speed, bool flowctl)
 	memset(&ti, 0, sizeof(ti));
 	cfmakeraw(&ti);
 
-	ti.c_cflag |= (speed | CLOCAL | CREAD);
+	ti.c_cflag |= (CLOCAL | CREAD);
+	if (cfsetspeed(&ti, speed) < 0) {
+		perror("Failed to set serial port speed");
+		close(fd);
+		return -1;
+	}
 
 	if (flowctl) {
 		/* Set flow control */
-- 
2.53.0

